.github/workflows/docker-singularity-publish.yml

```diff
@@ -0,0 +1,137 @@
+name: Docker and Singularity build
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+#  schedule:
+#    - cron: '31 0 * * *'
+  push:
+    branches: [ main ]
+    # Publish semver tags as releases.
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ main ]
```

Comment:
do we really want this to run on every PR?

Reply:
probably not, no - maybe just releases or the nightly cron?

Reply:
modified to "on demand"